dm/config/subtask_test.go

```diff
@@ -168,3 +170,33 @@ func (t *testConfig) TestSubTaskBlockAllowList(c *C) {
 	c.Assert(err, IsNil)
 	c.Assert(cfg.BAList, Equals, filterRules2)
 }
+
+func (t *testConfig) TestDBConfigClone(c *C) {
+	a := &DBConfig{
+		Host:     "127.0.0.1",
+		Port:     4306,
+		User:     "root",
+		Password: "123",
+		Session:  map[string]string{"1": "1"},
+		RawDBCfg: DefaultRawDBConfig(),
+	}
+
+	// When add new fields, also update this value
+	c.Assert(reflect.Indirect(reflect.ValueOf(a)).NumField(), Equals, 8)
+
+	b := a.Clone()
+	c.Assert(a, DeepEquals, b)
```

Comment:
Please also check that the inner pointer are point to different addresses

Reply:
https://github.com/pingcap/dm/pull/1955/commits/d2a4d6fcedb6c3e8d858a9c9bc89d9f0e026a9b6